cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -122,16 +122,40 @@ object InitialWorkDirFileGeneratorExpression {
 
     implicit val caseExpression: Case.Aux[Expression, InitialWorkDirFileEvaluator] = {
       at { expression =>
-        (_, unmappedParameterContext, mappedInputValues, expressionLib) => {
-          // A single expression which must evaluate to an array of Files
+        (ioFunctionSet, unmappedParameterContext, _, expressionLib) => {
+          // A single expression which should evaluate to a File or an array of Files.
           val expressionEvaluation = ExpressionEvaluator.eval(expression, unmappedParameterContext, expressionLib)
 
+          def stageFile(file: WomFile): Future[WomSingleFile] = {
+            // TODO WomFile could be a WomMaybePopulatedFile with secondary files but this code only stages in
+            // the primary file.
```

Comment:
OOC: Is this because it's super annoying to do or because we haven't seen a use case for it yet ?

Reply:
We definitely haven't seen a use case for it yet and I'm not 100% sure how to deal with directories tbh.